scripts/gulpfiles/test_tasks.js

```diff
@@ -31,84 +31,77 @@ const BOLD_GREEN = '\x1b[1;32m';
 const BOLD_RED = '\x1b[1;31m';
 const ANSI_RESET = '\x1b[0m';
 
-class Tester {
-  constructor(tasks = []) {
-    this.successCount = 0;
-    this.failCount = 0;
-    this.tasks = tasks;
-  }
-
-  /**
-   * Run all tests in sequence.
-   */
-  async runAll() {
-    for (const task of this.tasks) {
-      await this.runTestTask(task)
-    }
-    this.reportTestResult();
-  }
+let successCount = 0;
+let failCount = 0;
+let firstErr;
 
-  /**
-   * Create a Gulp task to run all tests.
-   */
-  asTask() {
-    return this.runAll.bind(this);
-  }
-
-  /**
-   * Run an arbitrary Gulp task as a test.
-   * @param {function} task Any Gulp task.
-   * @return {Promise} Asynchronous result.
-   */
-  async runTestTask(task) {
-    const id = task.name;
+/**
+ * Run an arbitrary Gulp task as a test.
+ * @param {function} task Any Gulp task.
+ * @return {Promise} Asynchronous result.
+ */
+async function runTestTask(id, task) {
```

Comment:
Does this still need to be async if we're explicitly returning a promise? (genuine question)

Reply:
No